--- a/tools/licenses/licenses.py
+++ b/tools/licenses/licenses.py
@@ -808,10 +808,16 @@ def _GnBinary():
   exe = 'gn'
   if sys.platform.startswith('linux'):
     subdir = 'linux64'
+    if os.path.exists(exe):
+      return './' + exe
+    return exe  # find in PATH
   elif sys.platform == 'darwin':
     subdir = 'mac'
+    if os.path.exists(exe):
+      return './' + exe
+    return exe  # find in PATH
   elif sys.platform == 'win32':
-    subdir, exe = 'win', 'gn.exe'
+    subdir, exe = os.path.join('..', 'out', 'Release', 'gn_build'), 'gn.exe'
   else:
     raise RuntimeError("Unsupported platform '%s'." % sys.platform)
 
